flow/surface_frame.h

```diff
@@ -26,6 +26,9 @@ class SurfaceFrame {
   using SubmitCallback =
       std::function<bool(SurfaceFrame& surface_frame, DlCanvas* canvas)>;
 
+  using DeferredSubmit = std::function<bool()>;
```

Comment:
@chinmaygarde this is something I could use some help designing a better solution for. 

Essentially what I need is to split SurfaceFrame submit into two methods, Flush and Submit (with the current Submit method being Flush+Submit). The flush would do everything up to presenting the drawable, while submit would actually do the presentation. That would allow us to finish the encoding work on the raster thread and then present from the platform thread as part of the CATransaction.

But this is complicated by all of the layers between SurfaceFrame and the impeller::Frame, and what to do about backends where a split Flush/Submit isn't feasible yet.

Reply:
We can sketch this offline but SurfaceFrame may be EOL of its use. How about a SurfaceFrameBuilder that creates a view hierarchy state representation that you can submit elsewhere?

Reply:
Walk me through it :)

Or throw some time on my calendar